src/gateway_api/network_antenna/network_antenna.rs

```diff
@@ -0,0 +1,17 @@
+use crate::prelude::*;
+
+/// Trait for executing network requests, to be  implemented on the FFI side
+/// (iOS/Android), so that Sargon can with some HTTP client perform make network
+/// requests.
+#[uniffi::export(with_foreign)]
+#[async_trait::async_trait]
+pub trait NetworkAntenna: Send + Sync {
+    /// Method invoked by Sargon Rust side, telling an implementing type to
+    /// execute a `NetworkRequest` and pass the Result back to Sargon Rust side.
+    ///
+    /// Either: `Err` or `Ok(NetworkResponse)`.
+    async fn execute_network_request(
```

Comment:
antenna moved to seperate file, documented and renamed method -> `execute_network_request`